src/Codeception/Lib/Generator/Actor.php

```diff
@@ -236,5 +235,45 @@ public function getNumMethods()
         return $this->numMethods;
     }
 
+    private function getDefaultValue(\ReflectionParameter $param)
+    {
+        $isPlainArray = function ($value) {
+            return ((count($value) === 0)
+                || (
+                    (array_keys($value) === range(0, count($value) - 1))
+                    && (0 === count(array_filter(array_keys($value), 'is_string'))))
+            );
+        };
+        if ($param->isDefaultValueAvailable()) {
+            if (method_exists($param, 'isDefaultValueConstant ') && $param->isDefaultValueConstant()) {
+                $constName = $param->getDefaultValueConstantName();
+                if (false !== strpos($constName, '::')) {
+                    list($class, $const) = explode('::', $constName);
+                    if (in_array($class, ['self', 'static'])) {
+                        $constName = $param->getDeclaringClass()->getName().'::'.$const;
+                    }
+                }
+
+                return $constName;
+            }
 
+            $paramValue = $param->getDefaultValue();
+            switch (true) {
```

Comment:
Wonder if it would be worth just to use one of the already existing php-encoder libraries.
I was playing around with `riimu/kit-phpencoder` seems to cover most cases.

If you are OK with using it, I can redo the PR with this library.